extendr-api/Cargo.toml

```diff
@@ -12,5 +12,13 @@ repository = "https://github.com/extendr/extendr"
 libR-sys = "0.1.12"
 extendr-macros = { path = "../extendr-macros", version="0.1.11" }
 extendr-engine = { path = "../extendr-engine", version="0.1.11" }
-ndarray = "0.13.1"
+ndarray = { version = "0.13.1", optional = true }
 
+[features]
+default = []
+
+# All features needed for testing
+tests-all = ["ndarray", "libR-sys/use-bindgen"]
+
+# All features needed for testing, minus ndarray
+tests-wo-ndarray = ["libR-sys/use-bindgen"]
```

Comment:
I think the questions is how we envision this in the future, when there are multiple optional modules. Presumably we wouldn't test all possible combinations of presence and absence but simply all optional modules present and all absent. This would suggest something like `tests-minimal`.

Alternatively, it may be a good idea to add a `use-bindgen` feature so people can turn it on without having to know that it is actually a feature of libR-sys.

Reply:
Makes sense. Let's use tests-minimal.

Reply:
Are we good to go with this merge?

Reply:
Yes, I believe this is ready. What I was a bit wondering is whether I should also add this, 

> Alternatively, it may be a good idea to add a `use-bindgen` feature so people can turn it on without having to know that it is actually a feature of libR-sys.

but let's discuss it when it's needed. Let's merge this.